autometa/config/databases.py

```diff
@@ -66,230 +97,504 @@ class Databases:
         missing = {}
-        for section in Databases.SECTIONS:
+        for section in sections:
             for option in self.config.options(section):
+                # Skip user added options not required by Autometa
                 if option not in Databases.SECTIONS.get(section):
-                    # Skip user added options not required by Autometa
                     continue
                 fpath = self.config.get(section, option)
-                if os.path.exists(fpath) and os.stat(fpath).st_size >= 0:
-                    # TODO: [Checkpoint validation]
-                    logger.debug(f'({section},{option}): {fpath}')
+                if os.path.exists(fpath):
                     continue
                 if validate:
-                    return False
+                    return True
                 if section in missing:
                     missing[section].add(option)
                 else:
-                    missing.update({section: set([option])})
-        for section, opts in missing.items():
-            for opt in opts:
-                logger.debug(f'MISSING: ({section},{opt})')
-        return True if validate else missing
+                    missing.update({section:set([option])})
+        # Log missing options
+        for section,options in missing.items():
+            for option in options:
+                logger.debug(f'MISSING: ({section},{option})')
+        # ReturnType based on provided `validate` parameter
+        if validate:
+            return False
+        else:
+            return missing
+
+    def download_missing(self, section=None):
+        """Download missing Autometa database dependencies from provided `section`.
+        If no `section` is provided will check all sections.
+
+        Parameters
+        ----------
+        section : str, optional
+            Section to check for missing database files (the default is None).
+            Choices include 'ncbi' and 'markers'.
+
+        Returns
+        -------
+        NoneType
+            Will update provided `section` in `self.config`.
+
+        Raises
+        -------
+        ValueError
+            Provided `section` does not match 'ncbi' and 'markers'.
+
+        """
+        dispatcher = {'ncbi':self.download_ncbi_files, 'markers':self.download_markers}
+        if section and section not in dispatcher:
+            raise ValueError(f'{section} does not match "ncbi" or "markers"')
+        if section:
+            missing = self.get_missing(section=section)
+            options = missing.get(section, [])
+            dispatcher[section](options)
+        else:
+            self.get_missing()
+            for section,options in missing.items():
+                dispatcher[section](options)
 
-    def update_missing(self):
-        """Download and format databases for all options in each section.
+    def compare_checksums(self, section=None, validate=False):
```

Comment:
I think we need a related function which looks at the checksum stored for a local file, then recalculates the file's checksum to see if they are the same. This could be used for files created by the pipeline itself to ensure the file is OK. If the md5 file is not there then that would signify that the step didn't finish properly.

Reply:
For example, this would be needed for the individual files that are untarred from taxdump. But it could also be applied to all other files made in other parts of the pipeline.

Reply:
I'm not sure how we would do these comparisons without having the remote checksum available. Otherwise, we can perform the checksum comparisons like I am doing with the database files. A general checksum comparison for files in the pipeline is likely to be a bit different because there will is not a remote checksum. `compare_checksums` here is specific to databases. Another checksum comparison function can be added to utilities to aid in comparing pipeline generated files.